src/select/util/selectTypes.ts

```diff
@@ -1,13 +1,14 @@
 import React from "react";
 
-interface Option {
-  id: string;
+interface Option<Id = string> {
+  id: Id;
+  title: React.ReactNode;
   isDisabled?: boolean;
 }
 
-interface TypeaheadSelectOption extends Option {
-  title: string;
-}
+type TypeaheadSelectOption<Id = string> = Omit<Option, "id"> & {
+  id: Id;
+};
```

Comment:
I guess that would also work
```suggestion
type TypeaheadSelectOption<Id = string> = Option<Id>;
```